tolstoy/src/metadata.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use TolstoyResult;
+use uuid;
+use schema;
+use rusqlite;
+use Result;
+
+trait SyncMetadataClientInterface {
```

Comment:
Traits are typically named via adjectives; in this case you might choose `HeadTrackable` or `MetadataStore`.